cranelift-simplejit/src/memory.rs

```diff
@@ -97,7 +97,12 @@ impl Memory {
     }
 
     /// TODO: Use a proper error type.
-    pub fn allocate(&mut self, size: usize) -> Result<*mut u8, String> {
+    pub fn allocate(&mut self, size: usize, align: u8) -> Result<*mut u8, String> {
+        if self.position % align as usize != 0 {
+            self.position += align as usize - self.position % align as usize;
```

Comment:
Is this correct?

Reply:
I think so, at least it matches what Spidermonkey [does](https://searchfox.org/mozilla-central/source/js/src/jsutil.h#156) (although in the Spidermonkey code there's an additional modulo in case `self.position` is 0 or a multiple of alignment, but the explicit check makes it obvious here).